src/pact.ts

```diff
@@ -56,6 +58,43 @@ export class Pact {
 		return q.all<Server>(_.map(this.__servers, (server:Server) => server.delete() as PromiseLike<Server>));
 	}
 
+	// Creates stub with specified options
+	public createStub(options: StubOptions = {}): Stub {
+		if (options && options.port && _.some(this.__stubs, (s: Stub) => s.options.port === options.port)) {
+			let msg = `Port '${options.port}' is already in use by another process.`;
+			logger.error(msg);
+			throw new Error(msg);
+		}
+
+		let stub = stubFactory(options);
+		this.__stubs.push(stub);
+		logger.info(`Creating Pact Stub with options: \n${this.__stringifyOptions(stub.options)}`);
+
+		// Listen to stub delete events, to remove from stub list
+		stub.once("delete", (s: Stub) => {
+			logger.info(`Deleting Pact Stub with options: \n${this.__stringifyOptions(s.options)}`);
+			this.__stubs = _.without(this.__stubs, s);
+		});
+
+		return stub;
+	}
+
+	// Return arrays of all stubs
+	public listStubs(): Stub[] {
+		return this.__stubs;
+	}
+
+	// Remove all the stubs that have been created
+	// Return promise of all others
+	public removeAllStubs(): q.Promise<Stub[]> {
```

Comment:
For simplicity in testing, I think we should add a `removeAll` function that simply removes all servers and stubs.